src/main/java/org/opensearch/knn/indices/ModelGraveyard.java

```diff
@@ -151,20 +153,54 @@ public static NamedDiff readDiffFrom(StreamInput streamInput) throws IOException
      * @throws IOException
      */
     public static ModelGraveyard fromXContent(XContentParser xContentParser) throws IOException {
+        // Added validation checks to validate all the different possible scenarios
+        // model_ids:"abcd"
+        // {}
+        // {["abcd", "1234"]}
+        // {"dummy_field_name":}
+        // {model_ids:"abcd"}
+        // {model_ids:null}
+        // {model_ids:[]}
+        // {model_ids: ["abcd", "1234"]}
+
         ModelGraveyard modelGraveyard = new ModelGraveyard();
 
         // If it is a fresh parser, move to the first token
         if (xContentParser.currentToken() == null) {
             xContentParser.nextToken();
         }
 
-        // on a start object move to next token
-        if (xContentParser.currentToken() == XContentParser.Token.START_OBJECT) {
-            xContentParser.nextToken();
+        // Validate if the first token is START_OBJECT
+        if (xContentParser.currentToken() != XContentParser.Token.START_OBJECT) {
+            throw new OpenSearchParseException(
+                "Unable to parse ModelGraveyard. Expecting START_OBJECT but got " + xContentParser.currentToken()
```

Comment:
Shouldn't we avoid to use START_OBJECT in the error message?

```suggestion
                "Unable to parse ModelGraveyard. Expecting a start of an object but got " + xContentParser.currentToken()
```

Reply:
I'm not sure about this. @navneet1v can you please suggest what is the best practice?

Reply:
@naveentatikonda what heemin suggested seems to be fine. Rather than saying "START_OBJECT" in error message which is a constant variable name its better to use `a start of an object` which seems to be fine.

Reply:
Got it, Thanks!